targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/serial_api.c

```diff
@@ -0,0 +1,314 @@
+ *   - Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   - Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *   - Modified versions of the software must be conspicuously marked as such.
+ *   - This software is licensed solely and exclusively for use with processors
+ *     manufactured by or for Analog Devices, Inc.
+ *   - This software may not be combined or merged with other code in any manner
+ *     that would cause the software to become subject to terms and conditions
+ *     which differ from those listed here.
+ *   - Neither the name of Analog Devices, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *   - The use of this software may or may not infringe the patent rights of one
+ *     or more patent holders.  This license does not release you from the
+ *     requirement that you obtain separate licenses from these patent holders
+ *     to use this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-
+ * INFRINGEMENT, TITLE, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL ANALOG DEVICES, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, DAMAGES ARISING OUT OF
+ * CLAIMS OF INTELLECTUAL PROPERTY RIGHTS INFRINGEMENT; PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+#include "serial_api.h"
+
+#if DEVICE_SERIAL
+
+// math.h required for floating point operations for baud rate calculation
+#include <math.h>
+#include "mbed_assert.h"
+
+#include <string.h>
+
+#include "cmsis.h"
+#include "pinmap.h"
+#include "PeripheralPins.h"
+#include "drivers/uart/adi_uart.h"
+#define ADI_UART_MEMORY_SIZE   (ADI_UART_BIDIR_MEMORY_SIZE)
+#define ADI_UART_NUM_DEVICES 2
+
+static ADI_UART_HANDLE          hDevice[ADI_UART_NUM_DEVICES];
+static uint32_t UartDeviceMem[ADI_UART_NUM_DEVICES][(ADI_UART_MEMORY_SIZE + 3)/4];
+static uint32_t serial_irq_ids[ADI_UART_NUM_DEVICES] = {0};
+static uart_irq_handler irq_handler = NULL;
+int stdio_uart_inited = 0;
+serial_t stdio_uart;
+int rxbuffer[2];
```

Comment:
why are these global? I can see these buffers are locally used in one function, could this be local scope then? not certain neither about why 2 are needed (based on the index), but if local scope would be, it would be just one character?

Reply:
The variables stdio_uart_inited and stdio_uart need to be global as they are referenced from mbed_board.c. The rxbuffer and txbuffers can be made static. As mentioned above, ADuCM4050 has two UARTs so it makes sense to provide code to support both even if only one is required for mbed-os.

Reply:
Understood. if possible, should be local then, or at least these should be made privately to this compilation unit (`static`).